test/cpp/end2end/ssl_credentials_test.cc

```diff
@@ -523,8 +517,7 @@ TEST_P(SslCredentialsTest, ClientCertificateIsUntrusted) {
           grpc_ssl_client_certificate_request_type::
               GRPC_SSL_REQUEST_AND_REQUIRE_CLIENT_CERTIFICATE_AND_VERIFY) {
     EXPECT_EQ(auth_context.status().code(), absl::StatusCode::kUnavailable);
-    EXPECT_THAT(auth_context.status().message(),
-                HasSubstr("CERTIFICATE_VERIFY_FAILED"));
+    EXPECT_FALSE(auth_context.status().message().empty());
```

Comment:
Why was this changed to not check the status message?  In general, it's better to check the full text of status messages in tests, since having useful messages for users is important for debuggability, so we do actually want tests to check that we're emitting useful messages.

Reply:
Greg noticed a small mistake in the test setup, which changed the error message. Specifically, the new client-side error message is (correctly) something that doesn't have any TLS-specific info it, it's just something like "failed to connect to all addresses: ... : Socket closed". Changed this back to check that the "Socket closed" substring is present (rather than checking the full error message, which I'm not confident will be stable).

Reply:
Why does the status message not include any TLS-specific info?  I thought I'd fixed that in #30077.  If not, then we still have some work to do here.

Hmm, is the problem here that the server's verification of the client's cert doesn't happen until after the handshake is done?  If so, that's really unfortunate, because this error message doesn't tell users anything about what the actual problem is.

Reply:
Your guess is correct: The server verifies the client cert after the client-side thinks the TLS handshake has finished, so the server-side logs this error (e.g. something with "CERTIFICATE_VERIFY_FAILED ...") and then the server closes the connection, so all that the client sees is that the connection was closed.

I agree that it's an unfortunate situation (other TLS stacks, e.g. Golang's crypto/tls, have the same issue). One idea would be to add some state to the "Socket closed" message (e.g. "Socket closed after TLS handshake completed and before received any application data from server"). Interested to hear your thoughts on this.

Reply:
Okay.  I don't see an immediate way to solve this, so let's not block this PR on it.  But please add a TODO here describing the problem and noting that we need to ultimately find a way to improve this error message for users.

Reply:
Done.